lib/sinon/util/fake_server.js

```diff
@@ -21,7 +21,29 @@ function responseArray(handler) {
     return response;
 }
 
-var wloc = typeof window !== "undefined" ? window.location : { "host": "localhost", "protocol": "http"};
+function getDefaultWindowLocation() {
+    return { "host": "localhost", "protocol": "http" };
+}
+
+function getWindowLocation() {
+    if ( typeof window !== "undefined") {
+        if (typeof window.location !== "undefined") {
+            // Browsers place location on window
+            return window.location;
```

Comment:
When you use return inside `if` statements, you can avoid using `else`.

See http://rmurphey.com/blog/2012/12/10/js-conditionals

Also, you can unfold this code a bit, so there are fewer levels of indentation == easier to understand.

```javascript
function getWindowLocation() {
    if (typeof window === "undefined") {
        // Fallback
        return getDefaultWindowLocation();
    }

    if (typeof window.location !== "undefined") {
        // Browsers place location on window
        return window.location;
    }

    if ((typeof window.window !== "undefined") && (typeof window.window.location !== "undefined")) {
        // React Native on Android places location on window.window
        return window.window.location;
    }

    return getDefaultWindowLocation();

}
```